concourse/steps/component_descriptor.mako

```diff
@@ -174,22 +178,28 @@ subprocess.run(
   env=subproc_env
 )
 
-# ensure the script actually created an output
-if not os.path.isfile(v2_outfile):
-  fail(f'no descriptor file was found at: {v2_outfile=}')
+if os.path.isdir(ctf_out_dir) and os.path.listdir(ctf_out_dir):
+  if os.path.isfile(v2_outfile):
+    fail('Both CTF dir and component-descriptor-v2-file found.')
+  else:
+    pass
+else:
+  # ensure the script actually created an output
+  if not os.path.isfile(v2_outfile):
+    fail(f'no descriptor file was found at: {v2_outfile=}')
```

Comment:
I think this check is redundant, as `parse_yaml_file` will also fail if the file be absent

Reply:
Strictly speaking that was not changed/introduced by this PR, but I'll refactor it

Reply:
On second thought - I prefer the more expressive error message.

Reply:
`parse_yaml_file` does call `existing_file`, does it not?

Reply:
so you refer to the `no descriptor file` part? ok, fine w/ me